keps/sig-windows/windows-runtimeclass-support.md

```diff
@@ -0,0 +1,575 @@
+
+Starting from story 2, the RuntimeClass `handler` field could also be used as a means os/version/architecture/variant used by the container runtime. The `handler` is matched up with a corresponding section in the ContainerD configuration file.
+
+Here's an example of what corresponding ContainerD configurations could look like for the runtimes above. The first one would force Hyper-V for compatibility with containers needing 10.0.17763. The second one would be the default for the current version where no Hyper-V isolation is required for compatibility.
+
+```toml
+        [plugins.cri.containerd.runtimes.containerd-hyperv-17763]
+          runtime_type = "io.containerd.runhcs.v1"
+          [plugins.cri.containerd.runtimes.containerd-hyperv-17763.options]
+            SandboxImage = "{{WINDOWSSANDBOXIMAGE}}"
+            SandboxPlatform = "windows/amd64"
+            SandboxOsVersion = "10.0.17763"
+            SandboxIsolation = 1
+
+        [plugins.cri.containerd.runtimes.default]
+          runtime_type = "io.containerd.runhcs.v1"
+          # No version is specified for process isolation, the node OS version is used
+```
+
+### Implementation Details/Notes/Constraints
+
+There were multiple options discussed with SIG-Node & SIG-Windows on October 8 2019 prior to filing this KEP. That discussion & feedback were captured in [Difficulties in mixed OS and arch clusters]. If you're looking for more details on other approaches excluded, please review that document.
+
+#### Adding new label node.kubernetes.io/windows-build
+
+In [Bounding Self-Labeling Kubelets], a specific range of prefixes were reserved for node self-labeling - `[*.]node.kubernetes.io/*`. Adding a new label within that namespace won't require any changes to NodeRestriction admission. As a new field it also won't require changes to any existing workloads.
+
+Build numbers will be used instead of product names in the node labels for a few reasons:
+
+- The same OS version may be marketed under two different names due to support / licensing differences. For example, Windows Server 2019 and Windows Server version 1809 are the same build number. The actual binary compatibility is based on build number and the same container can run on either.
+- Windows product names may not have been announced when the Kubernetes project contributors start building and testing against it. Using build numbers instead avoids needing to change Kubernetes source once a name has been announced.
+
+Here are the current product name to build number mappings to illustrate the point:
+
+|Product Name                          |   Build Number(s)      |
+|--------------------------------------|------------------------|
+| Windows Server 2019                  | 10.0.17763             |
+| Windows Server version 1809          | 10.0.17763             |
+| Windows Server version 1903          | 10.0.18362             |
+| Windows Server vNext Insider Preview | 10.0.18975, 10.0.18985 |
```

Comment:
Do we publish container images for `vNext`?

Reply:
Yes - see https://docs.microsoft.com/en-us/virtualization/windowscontainers/quick-start/using-insider-container-images#install-base-container-image

or `curl -L https://mcr.microsoft.com/v2/windows/servercore/insider/tags/list` for a list of all builds.

Reply:
I'm adding these links in the next revision for completeness  sake

Reply:
added "see [Container Base Images] and [Windows Insider Container Images]"